pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/signature/JavaFieldSignature.java

```diff
@@ -54,10 +55,10 @@ private static int code(Visibility visibility, boolean isStatic, boolean isFinal
      *
      * @return The signature of the field
      */
-    public static FieldSignature buildFor(ASTFieldDeclaration node) {
+    public static JavaFieldSignature buildFor(ASTFieldDeclaration node) {
         int code = code(Visibility.get(node), node.isStatic(), node.isFinal());
```

Comment:
I guess, the same applies here for potential collisions. In theory, it could be a problem. But in that case here, we anyway have a limited number of combinations (visibility, static, final) - so, I guess, we won't run into a collision...

This pooling behavior is worth a unit test, if we don't have one yet.